Submitted and accepted upstream.

=== modified file 'src/java/slashtime/client/Master.java'
--- src/java/slashtime/client/Master.java	2009-12-05 07:27:00 +0000
+++ src/java/slashtime/client/Master.java	2010-02-05 07:39:43 +0000
@@ -31,11 +31,16 @@
     public static UserInterface ui = null;
 
     public static void main(String[] args) {
+        boolean startMaximized = true;
+
         Glib.setProgramName("slashtime");
         Gtk.init(args);
         Internationalization.init("slashtime", "share/locale/");
 
         /*
+         * An optional first "--hidden" argument allows starting minimized
+         * in tray by not toggle() in ZonesWindow.initialPresentation().
+         *
          * If you specify a zone name on the command line it will become a
          * third icon in the ZonesWindow denoting where "home" is (as opposed
          * to where you are now). This is a bit of a hack at the moment, but
@@ -43,8 +48,14 @@
          * setting once we have a GConf binding.
          */
 
-        if (args.length == 1) {
-            specifyHome(args[0]);
+        for (String s : args) {
+            if ("--hidden".compareTo(s) == 0) {
+                startMaximized = false;
+                continue;
+            }
+
+            specifyHome(s);
+            break;
         }
 
         /*
@@ -52,7 +63,7 @@
          * re-entry point
          */
 
-        ui = new UserInterface();
+        ui = new UserInterface(startMaximized);
 
         /*
          * And, fire the main event loop.

=== modified file 'src/java/slashtime/ui/UserInterface.java'
--- src/java/slashtime/ui/UserInterface.java	2009-12-05 07:27:00 +0000
+++ src/java/slashtime/ui/UserInterface.java	2010-02-05 07:39:43 +0000
@@ -41,15 +41,15 @@
      * this would be constructed once, by the client layer, and assigned to
      * Master.ui
      */
-    public UserInterface() {
+    public UserInterface(boolean show) {
         loadImages();
         setupApplication();
-        setupWindows();
+        setupWindows(show);
     }
 
-    private void setupWindows() {
+    private void setupWindows(boolean show) {
         meeting = null;
-        zones = new ZonesWindow();
+        zones = new ZonesWindow(show);
         status = new DockedIndicator();
     }
 

=== modified file 'src/java/slashtime/ui/ZonesWindow.java'
--- src/java/slashtime/ui/ZonesWindow.java	2008-11-25 11:33:58 +0000
+++ src/java/slashtime/ui/ZonesWindow.java	2010-02-05 07:39:43 +0000
@@ -74,6 +74,8 @@
      */
     private boolean up = false;
 
+    private boolean showOnStartup;
+
     private VBox top;
 
     private TreeView view;
@@ -111,7 +113,9 @@
     /**
      * Build the main GUI window
      */
-    ZonesWindow() {
+    ZonesWindow(boolean show) {
+        showOnStartup = show;
+
         setupWindow();
         setupTreeView();
         setupContextMenu();
@@ -685,7 +689,9 @@
          * Toggle the ZonesWindow onto the screen. Among other things, this
          * will size, and present.
          */
-        toggle();
+        if (showOnStartup) {
+            toggle();
+        }
 
         // has to be after map to screen
         selection.unselectAll();

